src/sql/parts/modelComponents/editor.component.ts

```diff
@@ -57,17 +58,31 @@ export default class EditorComponent extends ComponentBase implements IComponent
 		this._editor = this._instantiationService.createInstance(QueryTextEditor);
 		this._editor.create(this._el.nativeElement);
 		this._editor.setVisible(true);
-		this._editorInput = this._instantiationService.createInstance(UntitledEditorInput, URI.from({ scheme: Schemas.untitled, path: `${this.descriptor.type}-${this.descriptor.id}` }), false, 'sql', '', '');
+		let uri = this.createUri();
+		this._editorInput = this._instantiationService.createInstance(UntitledEditorInput, uri, false, 'sql', '', '');
 		this._editor.setInput(this._editorInput, undefined);
 		this._editorInput.resolve().then(model => this._editorModel = model.textEditorModel);
 
 		this._register(this._editor);
 		this._register(this._editorInput);
 		this._register(this._editorModel.onDidChangeContent(e => {
 			this.content = this._editorModel.getValue();
+
+			// Notify via an event so that extensions can detect and propagate changes
+			this._onEventEmitter.fire({
+				eventType: ComponentEventType.onDidChange,
+				args: e
+			});
 		}));
 	}
 
+	private createUri(): URI {
```

Comment:
do we really need all of these uri to be set here? why are there so many?

Reply:
So... this is fun. Basically properties always overwrite from the client, so we need 2 to ensure that we track this correctly. I tried a few ways to shorten this but... nope, not with current design.